tools/hive-bigquery/gcs_storage_component.py

```diff
@@ -0,0 +1,211 @@
+        logger.debug("File %s doesn't exist", gcs_uri)
+        return False
+
+    def stage_to_gcs(self, mysql_component, bq_component, hive_table_model,
+                     bq_table_model, gcs_bucket_name):
+        """Copies staged files to GCS
+
+        Queries the tracking table, fetches information about the files to
+        copy to GCS, runs a distcp job to copy multiple files, and checks
+        whether the files have been successfully copied. If copied
+        successfully, updates the gcs_copy_status to 'DONE' else retries
+        copying.
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            bq_component (:class:`BigQueryComponent`): Instance of
+                BigQueryComponent to do BigQuery operations
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+            gcs_bucket_name (str): GCS bucket name
+        """
+
+        logger.debug(
+            "Fetching information about files to copy to GCS from tracking "
+            "table...")
+        select_query = "SELECT table_name,file_path FROM %s WHERE " \
+                       "gcs_copy_status='TODO'" % (
+                           hive_table_model.tracking_table_name)
+        results = mysql_component.execute_query(select_query)
+
+        if not results:
+            logger.debug("No file paths to copy to GCS")
+
+        while results:
+            file_info = {}
+            for row in results:
+                source_location = row[1]
+                file_name = source_location.split('/')[-1]
+                if file_name not in file_info.keys():
+                    file_info[file_name] = source_location
+            source_locations = ' '.join(file_info.values())
+            filename = "file_info.json"
+            # Dictionary of file names and their locations
+            with open(filename, "w") as file_content:
+                file_content.write(str(file_info))
+
+            target_blob = "BQ_staging/{}/{}/{}/".format(
+                hive_table_model.database_name,
+                hive_table_model.table_name.lower(), get_random_string())
+            # Uploads file to create a folder like structure in GCS
+            self.upload_file(gcs_bucket_name, filename, target_blob + filename)
+            os.remove(filename)
+
+            target_folder_location = "gs://{}/{}".format(gcs_bucket_name,
+                                                         target_blob)
+
+            logger.debug(
+                "Copying data from location %s to GCS Staging location %s "
+                "....", source_locations, target_folder_location)
+            # Hadoop distcp command to copy multiple files in one operation
+            cmd_copy_gcs = ['hadoop', 'distcp'] + file_info.values() + [

```

Comment:
why do we need to split the different parts of the command into multiple lists and then concat them. Can we not have one list instead?